modules/database/src/ioc/misc/afterInit.c

```diff
@@ -0,0 +1,126 @@
+/* Copyright (C) 2020 Dirk Zimoch */
+/* Copyright (C) 2020-2024 European Spallation Source, ERIC */
+
+#include <dbAccess.h>
+#include <epicsExport.h>
+#include <epicsStdio.h>
+#include <epicsString.h>
+#include <errlog.h>
+#include <errno.h>
+#include <initHooks.h>
+#include <iocsh.h>
+#include <stdlib.h>
+#include <string.h>
+
+// Version within the message
+static const char helpMessage[] =
+  "afterInit version 1.0.1\n"
+  "Allows you to define commands to be run after the iocInit\n"
+  "Example commands:\n"
+  "  afterInit \"dbpf <PV> <VAL>\"\n"
+  "  afterInit \"date\"\n";
+
+struct cmditem
+{
+  struct cmditem *next;
+  char *cmd;
+};
+
+static struct cmditem *cmdlist, **cmdlast = &cmdlist;
+
+static void afterInitHook(initHookState state)
+{
+  if(state != initHookAfterIocRunning)
+    return;
+
+  struct cmditem *item = cmdlist;
+  struct cmditem *next = NULL;
+  while(item)
```

Comment:
fyi. with `ellLib.h` this could become

```c
struct cmditem *item;
while((item = ellGet(theList)) {
```

Also, an AtShutdown hook should iterate in the opposite order.